Final_Project/FrameHandler.cpp

```diff
@@ -62,6 +62,14 @@ void receiveFrame(const char* frame) {
 void sendFrame(char* frame, const char* data, char ctrl) {
 	(ctrl != NULL) ? generateCtrlFrame(frame, ctrl)
 		: generateDataFrame(frame, data);
+
+	if (curState == "SEND" && ctrl == EOT) {
+		curState = "IDLE";
+		if (unfinishedTransmission) { // check whether everything twas sent
```

Comment:
I think this is almost right. Instead of checking `unfinishedTransmission` though, I would set it to true here for now. That way, in idle, we can check to see if `unfinishedTransmission` is true and bid for the line again.